File: Examples/VisualBasic/ProgrammersGuide/Working-with-User-defined-Cells/ReadUserdefinedCellsOfShape.vb
Imports VisualBasic
Imports Aspose.Diagram
Imports System

Public Class ReadUserdefinedCellsOfShape
    Public Shared Sub Run()
        'ExStart:ReadUserdefinedCellsOfShape
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_UserDefinedCells()

        ' load source Visio diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
        ' get page by name
        Dim page As Page = diagram.Pages.GetPage("Page-1")
        ' get shape by id
        Dim shape As Shape = page.Shapes.GetShape(1)
        ' extract user defined cells of the shape
        For Each user As User In shape.Users
            Console.WriteLine(user.Name + ": " + user.Value.Val)
        Next
        'ExEnd:ReadUserdefinedCellsOfShape
    End Sub
End Class

File: Examples/CSharp/Working-with-User-defined-Cells/ReadUserdefinedCellsOfShape.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_with_User_defined_Cells
{
    public class ReadUserdefinedCellsOfShape
    {
        public static void Run() 
        {
            // ExStart:ReadUserdefinedCellsOfShape
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_UserDefinedCells();

            // Load source Visio diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");
            // Get page by name
            Page page = diagram.Pages.GetPage("Page-1");
            // Get shape by id
            Shape shape = page.Shapes.GetShape(1);
            // Extract user defined cells of the shape
            foreach (User user in shape.Users)
            {
                Console.WriteLine(user.Name + ": " + user.Value.Val);
            }
            // ExEnd:ReadUserdefinedCellsOfShape
        }
    }
}
